compiler/resolution/lowerIterators.cpp

```diff
@@ -2509,10 +2514,16 @@ void lowerIterators() {
 
   forv_Vec(FnSymbol, fn, gFnSymbols) {
     if (fn->isIterator()) {
+     if (fn->firstSymExpr() == NULL &&
```

Comment:
I think this section needs a comment describing the general goal of this snippet, like "Remove parallel iterators that are not used"